app/authentication/roles.py

```diff
@@ -1,12 +1,13 @@
 from functools import wraps
+from typing import Any
 
 from flask_login import current_user
 from werkzeug.exceptions import Forbidden
 
 from app.globals import get_metadata
 
 
-def role_required(role):
+def role_required(role: str) -> Any:
```

Comment:
Decorators don't have a great support for typing. They should get better in 3.10.
But for now, this can be a little more specific.
```suggestion
def role_required(role: str) -> Callable:
```